GVRf/Framework/src/org/gearvrf/GVRMonoscopicViewManager.java

```diff
@@ -0,0 +1,157 @@
+
+
+    /**
+     * Constructs GVRMonoscopicViewManager object with GVRScript which controls GL
+     * activities
+     * 
+     * @param gvrActivity
+     *            Current activity object
+     * @param gvrScript
+     *            {@link GVRScript} which describes
+     * @param distortionDataFileName
+     *            distortion filename under assets folder
+     */
+    GVRMonoscopicViewManager(GVRActivity gvrActivity, GVRScript gvrScript,
+            String distortionDataFileName) {
+        super(gvrActivity, gvrScript, distortionDataFileName);
+
+        /*
+         * Sets things with the numbers in the xml.
+         */
+        GVRXMLParser xmlParser = new GVRXMLParser(gvrActivity.getAssets(),
+                distortionDataFileName);
+        
+        mView = new GVRSurfaceView(gvrActivity, this, null);
+        gvrActivity.setContentView(mView);
+
+        DisplayMetrics metrics = new DisplayMetrics();
+        gvrActivity.getWindowManager().getDefaultDisplay().getMetrics(metrics);
+
+        final float INCH_TO_METERS = 0.0254f;
+        int screenWidthPixels = metrics.widthPixels;
+        int screenHeightPixels = metrics.heightPixels;
+        float screenWidthMeters = (float) screenWidthPixels / metrics.xdpi
+                * INCH_TO_METERS;
+        float screenHeightMeters = (float) screenHeightPixels / metrics.ydpi
+                * INCH_TO_METERS;
+
+        mLensInfo = new GVRLensInfo(screenWidthPixels, screenHeightPixels,
+                screenWidthMeters, screenHeightMeters, xmlParser);
+
+        GVRPerspectiveCamera.setDefaultFovY(xmlParser.getFovY());
+        int fboWidth = xmlParser.getFBOWidth();
+        int fboHeight = xmlParser.getFBOHeight();
+        if(fboWidth <= 0) {
+            fboWidth = screenWidthPixels;
+        }
+        if(fboHeight <=0) {
+            fboHeight = screenHeightPixels;
+        }
+        float aspect = (float) fboWidth / (float) fboHeight;
+        GVRPerspectiveCamera.setDefaultAspectRatio(aspect);
+        mViewportX = 0;
+        mViewportY = 0;
+        mViewportWidth = fboWidth;
+        mViewportHeight = fboHeight;
+        if(fboWidth != screenWidthPixels) {
+            mViewportX = (screenWidthPixels / 2) - (fboWidth /2);
+        }        
+        if(fboHeight != screenHeightPixels) {
+            mViewportY = (screenHeightPixels / 2) - (fboHeight / 2);
+        }
+        
+    }
+
+    /*
+     * GL life cycle
+     */
+    @Override
+    void onDrawFrame() {
+        //Log.v(TAG, "onDrawFrame");
+        mFrameHandler.beforeDrawEyes();
+        mFrameHandler.onDrawFrame();
+        mFrameHandler.afterDrawEyes();
+    }
+
+    @Override
+    protected void drawEyes() {
+        //Log.d(TAG, "drawEyes()");
+        mMainScene.getMainCameraRig().predict(3.5f / 60.0f);
+        GVRMonoscopicRenderer.renderCamera(mMainScene, mMainScene.getMainCameraRig().getLeftCamera(),
+                mViewportX, mViewportY, mViewportWidth, mViewportHeight, mRenderBundle);
```

Comment:
Should we use a "center camera" here (as we did for screenshot-center), instead of left camera? There are at least two differences between center camera and left camera:
1. Center camera has slightly different view direction, which can approximate what the user see through GearVR in stereo mode;
2. Center camera can has different render mask so that all objects are visible. (If you run GVRSample in mono mode now, you do not see the right face because it is not visible in left camera.)